core/trino-spi/src/main/java/io/trino/spi/block/DictionaryBlock.java

```diff
@@ -212,10 +212,9 @@ private void calculateCompactSize()
         boolean[] used = new boolean[dictionary.getPositionCount()];
         for (int i = 0; i < positionCount; i++) {
             int position = getId(i);
-            if (!used[position]) {
-                uniqueIds++;
-                used[position] = true;
-            }
+            // Avoid branch mispredictions by setting of position unconditionally.
+            uniqueIds += used[position] ? 0 : 1;
```

Comment:
I don't think it converts to a simple cast, you need xor loginc. cc @skrzypo987

Reply:
You need `var ? 1: 0` for this to work. So in this case `(var ? 1 : 0) ^ 1